libs/application/templates/children-residence-change/src/fields/Overview/index.tsx

```diff
@@ -56,21 +75,88 @@ const Overview = ({ application }: FieldBaseProps) => {
     createPdfPresignedUrl,
     getPresignedUrl,
     application.state,
+    pdfType,
   ])
 
   const pdfUrl =
     createResponse?.createPdfPresignedUrl?.url ||
     getResponse?.getPresignedUrl?.url
 
+  setBeforeSubmitCallback &&
+    setBeforeSubmitCallback(async () => {
+      if (!pdfUrl) {
+        return [false, 'no pdf url']
+      }
+      dispatchFileSignature({ type: FileSignatureActionTypes.REQUEST })
+      const documentToken = await requestFileSignature({
+        variables: {
+          input: {
+            id: application.id,
+            type: pdfType,
+          },
+        },
+      })
+        .then((response) => {
+          return response.data?.requestFileSignature?.documentToken
+        })
+        .catch((error) => {
+          dispatchFileSignature({
+            type: FileSignatureActionTypes.ERROR,
+            status: FileSignatureStatus.REQUEST_ERROR,
+            error: '500',
+          })
+          throw new Error(`Request signature error ${JSON.stringify(error)}`)
+        })
+      if (documentToken) {
+        dispatchFileSignature({ type: FileSignatureActionTypes.UPLOAD })
+        const success = await uploadSignedFile({
+          variables: {
+            input: {
+              id: application.id,
+              documentToken: documentToken,
+              type: pdfType,
+            },
+          },
+        })
+          .then(() => {
+            return true
```

Comment:
We could return response.data?.uploadSignedFile?.documentSigned here, if we want too :) 
Think it makes the code more explicit of what we want from the endpoint.

Reply:
I agree and we should change this in another PR 😃